src/NearOrg/Notifications/NotificationsList.jsx

```diff
@@ -0,0 +1,274 @@
+  Math.max(initialRenderLimit + addDisplayCount * 2, index.options.limit ?? 0),
+  100
+);
+const reverse = !!props.reverse;
+
+let initialItems = Social.index(index.action, index.key, index.options);
+
+if (initialItems === null) {
+  return "";
+}
+const initialFoundItems = !!initialItems.length;
+// moderate
+initialItems = initialItems.filter((i) => !shouldFilter(i));
+
+const computeFetchFrom = (items, limit, previouslyFoundItems) => {
+  // we must get an explicit bool on whether we previously found items
+  // in order to determine whether to do the next fetch since we can't
+  // rely on the previous fetched count being less than the limit with
+  // moderation now in the picture
+  if (!previouslyFoundItems) {
+    return false;
+  }
+
+  const blockHeight = items[items.length - 1].blockHeight;
+  return index.options.order === "desc" ? blockHeight - 1 : blockHeight + 1;
+};
+
+const mergeItems = (newItems) => {
+  const items = [
+    ...new Set(
+      [...newItems, ...(state.items || [])].map((i) => JSON.stringify(i))
+    ),
+  ].map((i) => JSON.parse(i));
+  items.sort((a, b) => a.blockHeight - b.blockHeight);
+  if (index.options.order === "desc") {
+    items.reverse();
+  }
+  return items;
+};
+
+const jInitialItems = JSON.stringify(initialItems);
+if (state.jInitialItems !== jInitialItems) {
+  const jIndex = JSON.stringify(index);
+  if (jIndex !== state.jIndex) {
+    State.update({
+      jIndex,
+      jInitialItems,
+      items: initialItems,
+      fetchFrom: false,
+      nextFetchFrom: computeFetchFrom(
+        initialItems,
+        index.options.limit,
+        initialFoundItems
+      ),
+      displayCount: initialRenderLimit,
+      cachedItems: {},
+    });
+  } else {
+    State.update({
+      jInitialItems,
+      items: mergeItems(initialItems),
+    });
+  }
+}
+
+if (state.fetchFrom) {
+  const limit = addDisplayCount;
+  let newItems = Social.index(
+    index.action,
+    index.key,
+    Object.assign({}, index.options, {
+      from: state.fetchFrom,
+      subscribe: undefined,
+      limit,
+    })
+  );
+  if (newItems !== null) {
+    const newFoundItems = !!newItems.length;
+    // moderate
+    newItems = newItems.filter((i) => !shouldFilter(i));
+    State.update({
+      items: mergeItems(newItems),
+      fetchFrom: false,
+      nextFetchFrom: computeFetchFrom(newItems, limit, newFoundItems),
+    });
+  }
+}
+
+const makeMoreItems = () => {
+  State.update({
+    displayCount: state.displayCount + addDisplayCount,
+  });
+  if (
+    state.items.length - state.displayCount < addDisplayCount * 2 &&
+    !state.fetchFrom &&
+    state.nextFetchFrom &&
+    state.nextFetchFrom !== state.fetchFrom
+  ) {
+    State.update({
+      fetchFrom: state.nextFetchFrom,
+    });
+  }
+};
+
+const loader = (
+  <div className="loader" key={"loader"}>
+    <span
+      className="spinner-grow spinner-grow-sm me-1"
+      role="status"
+      aria-hidden="true"
+    />
+    Loading ...
+  </div>
+);
+
```

Comment:
@marcinbodnar can you please explain a bit why we need to query and build a feed in a similar way as `IndexFeed` and how does this fit into the push notification data stream?